### DISCUSSION TITLE
Changing quarkus.smallrye-openapi.servers when quarkus.swagger-ui.always-include=true

### DISCUSSION THREAD

**Discussion by edeandrea** [quarkusio/quarkus#26124] [Q&A] [answered]
It doesn't seem possible to change the value of `quarkus.smallrye-openapi.servers` if I have `quarkus.swagger-ui.always-include=true` (its documented as a build-time only property).

For example, when `quarkus.swagger-ui.always-include=true` and I deploy the app to kubernetes, I'd like to be able to override `quarkus.smallrye-openapi.servers` with the URL to the app.

Right now it defaults to `localhost` and the swagger-ui doesn't work when I try to use it.

![image](https://user-images.githubusercontent.com/363447/173614505-3c2ae3c5-2ddc-4285-bc0a-c4e5727b0d8f.png)

[a comment by an automated account is not shown]

**Comment by phillip-kruger** [comment, marked as answer]
Do you have a small reproducer?

**Reply by edeandrea** [reply]
1. `git clone https://github.com/quarkusio/quarkus-super-heroes.git`
2. `cd quarkus-super-heroes/rest-villains`
3. Log into OpenShift cluster (the [Red Hat Developer Sandbox](https://red.ht/dev-sandbox) works fine): `oc login ...`
4. Create a project: `oc new-project some-project`
5. Deploy to cluster (Java 11): `./mvnw clean package -Dquarkus.kubernetes.deploy=true -DskipTests -Dquarkus.kubernetes-client.trust-certs=true -Dquarkus.kubernetes.deployment-target=openshift -Dquarkus.container-image.builder=openshift`

   > If you want to use Java 17, you'll need to use this command: `./mvnw clean package -Dquarkus.kubernetes.deploy=true -DskipTests -Dquarkus.kubernetes-client.trust-certs=true -Dquarkus.kubernetes.deployment-target=openshift -Dquarkus.container-image.builder=openshift -Dquarkus.docker.dockerfile-jvm-path=src/main/docker/Dockerfile.jvm17 -Dquarkus.openshift.jvm-dockerfile=src/main/docker/Dockerfile.jvm17 -Dquarkus.openshift.base-jvm-image=registry.access.redhat.com/ubi8/openjdk-17:1.11`
7. Open Swagger UI: http://rest-villains-some-project.<YOUR_CLUSTER_URL>/q/swagger-ui

![image](https://user-images.githubusercontent.com/363447/173616923-16f5297f-bbf8-4a13-92d8-2e8836c63c17.png)

You can see that the swagger-ui is attempting to hit http://localhost:8084, which obviously isn't going to work.

**Reply by edeandrea** [reply]
@phillip-kruger I could probably put together instructions using minikube if that would be more helpful. Just let me know.

**Reply by phillip-kruger** [reply]
Can this not be recreated with Java -jar ? Why would kubernetes be special ?

**Reply by edeandrea** [reply]
If you're just running `java -jar` then swagger ui will work fine won't it? Swagger-ui will point to http://localhost, which is where your app is running. Thats the problem - I can't seem to update swagger-ui at runtime to tell it the actual URL of the application.

There is no way I can possibly know the URL of where the application is at build time.

**Reply by phillip-kruger** [reply]
But I mean if you set servers config when running java -jar, does that take affect?

**Reply by edeandrea** [reply]
> But I mean if you set servers config when running java -jar, does that take affect?

No

1. `git clone https://github.com/quarkusio/quarkus-super-heroes.git`
2. `cd quarkus-super-heroes/rest-villains`
3. `./mvnw clean package -DskipTests`
4. `java -Dquarkus.smallrye-openapi.servers=http://somewhere.else -jar target/quarkus-app/quarkus-run.jar`

```
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
12:17:28 WARN  [io.qu.ru.co.ConfigRecorder] (main) Build time property cannot be changed at runtime:
 - quarkus.smallrye-openapi.servers is set to 'http://somewhere.else' but it is build time fixed to 'http://localhost:8084'. Did you change the property quarkus.smallrye-openapi.servers after building the application?
```

**Reply by phillip-kruger** [reply]
Have you tried the mp config option, mp.openapi.servers ? See https://download.eclipse.org/microprofile/microprofile-open-api-1.1.2/microprofile-openapi-spec.html#_core_configurations

That should work with a -D type input

**Reply by edeandrea** [reply]
`mp.openapi.servers` does seem to work. Should it be a bug that `quarkus.smallrye-openapi.servers` does not?

**Reply by phillip-kruger** [reply]
I think that is a build time config . We can probably argue it should be a runtime one, so yea, an issue should help

**Reply by edeandrea** [reply]
I'll open one up. Thanks for the help!

**Reply by JohnneSouza** [reply]
So... there is an issue open for this one?

**Reply by edeandrea** [reply]
https://github.com/quarkusio/quarkus/issues/26136

**Reply by phillip-kruger** [reply]
This is now fixed. In the next release you should be able to the quarkus config
